src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarCreateAlertButton.tsx

```diff
@@ -70,30 +69,25 @@ const ArtworkSidebarCreateAlertButton: FC<ArtworkSidebarCreateAlertButtonProps>
     additionalGeneIDs,
   }
 
-  const getAuthModalOptions = (): AuthModalOptions => {
-    return {
-      entity: {
-        name: artwork.title!,
-        slug: artwork.slug,
-      },
-      afterSignUpAction: {
-        action: "createAlert",
-        kind: "artworks",
-        objectId: artwork.internalID,
-      },
-      contextModule: ContextModule.artworkSidebar,
-      intent: Intent.createAlert,
-      redirectTo: location.href,
-      copy: "Sign up to create your alert",
-    }
-  }
```

Comment:
I think this stuff was wrapped in a function because of the call on `window.location` — that's a default anyway so removing that minor bit of indirection. I'm not really convinced this is a good way to pass these options anyway but it's fine for now. Can revisit when we retire the legacy dialog.

Reply:
I think the _only_ thing you'd have to actually pass is the `contextModule` (which is what we do in a lot of these components). The rest can be inferred.